web/crux/src/interceptors/grpc.logger.interceptor.ts

```diff
@@ -14,17 +16,34 @@ export default class GrpcLoggerInterceptor implements NestInterceptor {
 
   constructor(private readonly helper: InterceptorGrpcHelperProvider) {}
 
+  shouldLogStack(err: Error) {
+    const grpcErr = err as BaseGrpcException
+    if (grpcErr.getError && (grpcErr.getError() as GrpcError).code !== Status.INTERNAL) {
+      return false
+    }
+    return true
+  }
+
   intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
     const result = this.helper.mapToGrpcObject(context)
     const data = JSON.stringify(result.data)
 
     this.logger.verbose(`gRPC ${result.serviceCall} called with the following object: ${data}`)
 
     return next.handle().pipe(
-      catchError((err: Error) => {
-        this.logger.error(`gRPC ${result.serviceCall} failed with: ${data}`, err.stack)
+      catchError((err: Error | BaseGrpcException) => {
+        const message = `gRPC ${result.serviceCall} failed with: ${data}`
+        if (this.shouldLogStack(err)) {
```

Comment:
if can be simplified to `this.logger.error(message, this.shouldLogStack(err) ? err.stack : undefined)`

Reply:
Tried it but for some reason it just prints another line with the text "undefined".

Like this:
```
[Nest] 7352  - 2022. 11. 30. 9:00:04   ERROR [GrpcLoggerInterceptor] gRPC /crux.CruxNode/SubscribeNodeEventChannel failed with: {"id":"78421c11-0228-4a6c-b1c1-92821eeb1aec"}
[Nest] 7352  - 2022. 11. 30. 9:00:04   ERROR [GrpcLoggerInterceptor] undefined
```

That's why I did it this way, but I'll see if I can do something.

Reply:
oh i see, that's because the error() func is overloaded and it picks the other one. My bad.